fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/CompletionExceptionLogHelper.java

```diff
@@ -55,6 +56,23 @@ public static void setAddSuppressed(boolean addSuppressed) {
         CompletionExceptionLogHelper.addSuppressed = addSuppressed;
     }
 
+    /**
+     * Change the maximum number of suppressed exceptions to add to any given exception, if this maximum has not already
+     * been set. Return whether or not the maximum was changed.
+     *
+     * This method only changes behavior when {@link #addSuppressed} is {@code true}.
+     * @param count the new maximum count
+     * @return {@code true} if the count was changed and {@code false} if it was not
+     */
+    public static synchronized boolean setMaxSuppressedCountIfNotSet(int count) {
```

Comment:
I think this will make it difficult to test different values of `count` in the unit tests, unless there's like a package-private visible-for-testing way of setting the count to something else.

Reply:
Yes, that's right. I added a scarily-named `@VisibleForTesting` method to allow this kind of testing and some tests to exercise it.